autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -125,26 +127,16 @@ def fit(self, X_train=None, Y_train=None, X_test=None, Y_test=None, dataset_trai
     def predict_proba(self, X, preprocess=True):
```

Comment:
Add argument: `normalize_pred_probas=None`

```
if normalize_pred_probas is None:
    normalize_pred_probas = self. normalize_pred_probas
```

This enables us to toggle this on during training if stopping_metric requires normalization. We wouldn't have to change stopping_metric to roc_auc in NN.

Reply:
Ideally we want to move this function to be _predict_proba, with predict_proba having the normalize_pred_probas arg to avoid duplicate code which is starting to pop up. predict_proba should only be defined in abstract_model, and individual models can implement _predict_proba if they have custom logic.

```
def predict_proba(self, X, preprocess=True, normalize=None):
    if normalize is None:
         normalize = self.normalize_pred_probas
    y_pred_proba = self._predict_proba(X=X, preprocess=preprocess)
    if normalize:
        y_pred_proba = normalize_pred_probas(y_pred_proba, self.problem_type)
    return y_pred_proba
```

With this, I don't think any changes to the model predict_proba functions are necessary.

Reply:
done